Automattic/newspack-theme
# Update theme/folder structure for move to parent/child themes

### All Submissions:


### Changes proposed in this Pull Request:

This PR updates the theme folder structure to nest the actual Newspack Theme inside of a /newspack-theme directory, and adjusts the build processes to generate the release zip into a different location, and add a couple new build processes that can be built out to include child themes as they're added. 

Related #594.

### How to test the changes in this Pull Request:

**Updated Dec 16** to improve testing instructions.

1. Clone the repository directly into the `/themes` folder using `git clone git@github.com:Automattic/newspack-theme.git .`. You will need to empty the directory first; if you still get the error `fatal: destination path '.' already exists and is not an empty directory.`, you can check for hidden files using `ls -a`.
Alternatively, you can clone the repo into a location separate from your themes directory, and symlink the nested 'newspack-theme' directory to a wp-content/themes directory on your test site. 
2. Apply the PR.
3. In the root of the directory, run `npm install && composer install`.
4. In the same location, try running `npm run build` and `npm run release:archive`. 
5. Confirm the theme is running as expected on your site.
6. Confirm that a zip has been generated in the `/releases` directory. 

Note: with the current structure, you can only run the build from the root directory, not from within the theme (since each of those directories don't have dependencies installed). It would work to install dependencies there, too, but we probably do want to build _all_ the themes every time we need to build the themes, so the current setup reflects the ideal processes. 

### Other information:
